curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f5d2776c854e069560fbe6.md

```diff
@@ -17,10 +17,10 @@ You should add four `label` elements.
 assert.equal(document.querySelectorAll('label')?.length, 4);
 ```
 
-You should add the `label` elements to the first `fieldset`.
+You should add the `label` elements as direct children to the first `fieldset`. Make sure to close the `label` elements.
 
 ```js
-assert.equal(document.querySelector('fieldset')?.querySelectorAll('label')?.length, 4);
+assert.equal(document.querySelectorAll('fieldset > label')?.length, 4);
```

Comment:
True, I guess I didn't consider that. Probably worse, or just bad in a different way.

Maybe we can use `fieldset:first-child > label`

I'm having a hard time thinking of all the different ways this can be written incorrectly.

Reply:
The test in place now also allows this.

```
<fieldset>
  <label></label>
  <label></label>
  <label></label>
  <label></label>
</fieldset>
<form method="post" action='https://register-demo.freecodecamp.org'>
  <fieldset></fieldset>
  <fieldset></fieldset>
</form>
```

Reply:
On one hand, it is unreasonable (_impossible_) to test for all incorrect ways to add the code.

On the other hand, we should probably check all the code within the editable region is as expected. So,

> Do we need to count the fieldset elements?

Probably. So, to close the issue, let us start from the top, and assert down:
1. There should only be 3 fieldset elements total
2. The fieldsets should all be direct children of the form
3. The first fieldset should have 4 label elements
4. The other fieldset elements should be empty

Reply:
We could also regex the closing elements. Something like:

```js
assert.lengthOf(document.querySelector('fieldset').textContent.match(/<\/label>/g), 4)
```